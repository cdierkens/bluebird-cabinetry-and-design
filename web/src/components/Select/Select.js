import PropTypes from "prop-types";
import React from "react";
import styles from "./Select.module.css";

const Option = ({ children }) => (
  <option className={styles.Option}>{children}</option>
);

Option.propTypes = {
  children: PropTypes.node,
};

const Select = ({ children }) => (
  <div className={styles.SelectWrapper}>
    <select className={styles.Select}>{children}</select>
    <div className={styles.Arrow}>
      <svg
        className="fill-current h-4 w-4"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 20 20"
      >
        <path d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z" />
      </svg>
    </div>
  </div>
);

Select.propTypes = {
  children: PropTypes.arrayOf(Option),
};

Select.Option = Option;

export default Select;
