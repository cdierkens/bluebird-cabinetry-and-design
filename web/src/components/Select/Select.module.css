.SelectWrapper {
  @apply block mb-2 relative pb-5 mb-2;
}

.Select {
  @apply block appearance-none w-full bg-white border border-gray-light px-3 py-3 pr-8 leading-tight font-sans text-base shadow-sm;
}

.Select:focus {
  @apply outline-none shadow-outline;
}

.Option {
  /* Empty */
}

.Arrow {
  @apply pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-darker;
}
