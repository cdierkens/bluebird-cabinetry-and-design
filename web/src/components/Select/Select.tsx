import { ErrorMessage, useField } from "formik";
import React from "react";
import * as styles from "./Select.module.css";

interface OptionProps {
  disabled?: boolean;
  value: string;
}

const Option: React.FC<OptionProps> = ({
  children,
  disabled = false,
  value,
}) => (
  <option value={value} className={styles.Option} disabled={disabled}>
    {children}
  </option>
);

interface SelectProps {
  label: string;
  name: string;
}

const Select: React.FC<SelectProps> & {
  Option: typeof Option;
} = ({ children, label, name }) => {
  const [field] = useField(name);

  return (
    <label className={styles.SelectWrapper}>
      <span className="block text-blue-dark text-base ml-1 mb-1">{label}</span>
      <div className="relative">
        <select className={styles.Select} {...field}>
          {children}
        </select>
        <div className={styles.Arrow}>
          <svg
            className="fill-current h-4 w-4"
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 20 20"
          >
            <path d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z" />
          </svg>
        </div>
      </div>
      <ErrorMessage name={field.name}>
        {(error) => (
          <span className="ml-1 text-sm absolute bottom-0 text-gold">
            {error}
          </span>
        )}
      </ErrorMessage>
    </label>
  );
};

Select.Option = Option;

export default Select;
