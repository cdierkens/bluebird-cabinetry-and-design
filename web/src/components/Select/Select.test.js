import { render } from "@testing-library/react";
import React from "react";
import Select from "./Select";

it("matches snapshot", () => {
  const { asFragment } = render(<Select label="mock-label" />);

  expect(asFragment()).toMatchSnapshot();
});
