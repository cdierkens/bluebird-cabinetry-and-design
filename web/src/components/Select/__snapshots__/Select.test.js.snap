// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SelectWrapper"
  >
    <select
      class="Select"
    />
    <div
      class="Arrow"
    >
      <svg
        class="fill-current h-4 w-4"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
